### PR TITLE
Adding support for SoxBindings.

### PR DESCRIPTION
**Pull request by pseeth**
This PR adds support for SoxBindings, if installed. It will also install SoxBindings on non-Windows computers. Windows computers will fall back to PySox.

There were some fixes that were needed in SoxBindings to make this work - I made the changes over there, and pushed a new release to PyPI: 1.2.0b0. Keeping it beta for now but after a bit I'll move it to 1.2.0 proper.

This PR also removes 2.7 and 3.4 from the test suite, and adds 3.8.

Regression data is untouched. I also added some lines to keep coverage at 100%, let me know what you think:

```
try:
    import soxbindings as sox 
except: # pragma: no cover
    import sox # pragma: no cover
```

The test suite happens only on a Linux machines so it uses SoxBindings in the test suite. SoxBindings and PySox are now equivalent with respect to Scaper. So I marked the lines importing PySox if importing SoxBindings fails as not needing to be covered.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 Changelog
 ---------
+v1.6.0
+~~~~~~
+- Uses soxbindings when installing on Linux or MacOS, which results in better performance.
+- Adds explicit support for Python 3.7 and 3.8.
```
Comment (justinsalamon) [inline docs/changes.rst:8]: Let's add a line to the changelog - "Drop support for python 2.7 and 3.4."

File: scaper/core.py [hunk 2]
```diff
@@ -1,4 +1,7 @@
-import sox
+try:
+    import soxbindings as sox 
+except: # pragma: no cover
+    import sox # pragma: no cover
```
Comment (justinsalamon) [inline scaper/core.py:4]: I think this is good for now, but let's create an issue so we remember that in the long term this needs to be addressed. It looks like GH actions supports running tests on all 3 OSs, so in an ideal scenario we can switch to GH actions and run the tests on all platforms, including Windows.

Screen Shot 2020-09-23 at 9 49 32 AM

**Review by justinsalamon — APPROVED** [review]
@pseeth lgtm let's just add a line to the changelog about dropping 2.7 and 3.4, and let's open an issue to remember we eventually want to add testing for Windows via GH actions. No need for a second CR.

**Comment by pseeth** [comment]
Cool, made those changes and released SoxBindings 1.2.0 (taking it out of beta). So should work now!

File: README.md [hunk 3]
```diff
@@ -23,30 +23,33 @@ In IEEE Workshop on Applications of Signal Processing to Audio and Acoustics (WA
...
-Scaper has two non-python dependencies:
-- SoX: http://sox.sourceforge.net/
+Scaper has one non-python dependency:
 - FFmpeg: https://ffmpeg.org/
 
+If you are installing Scaper on Windows, you will also need:
+- SoX: http://sox.sourceforge.net/
+
+If you are installing on Linux/macOS, the SoX dependency is taken care of via 
+[SoxBindings](https://github.com/pseeth/soxbindings).
```
Comment (justinsalamon) [inline README.md:33]: "On Linux/macOS SoX is replaced by [SoxBindings](https://github.com/pseeth/soxbindings) which is significantly faster, giving better overall performance in Scaper. On these platforms SoxBindings is installed automatically when calling `pip install scaper` (see below)."

File: docs/installation.rst [hunk 4]
```diff
@@ -5,19 +5,21 @@ Installation instructions
...
+Scaper has one non-python dependency:
+- FFmpeg: https://ffmpeg.org/
 
+If you are installing Scaper on Windows, you will also need:
 - SoX: http://sox.sourceforge.net/
-- FFmpeg: https://ffmpeg.org/
 
-On macOS these can be installed using `homebrew <https://brew.sh/>`_:
+If you are installing on Linux/macOS, the SoX dependency is taken care of via 
+[SoxBindings](https://github.com/pseeth/soxbindings).
```
Comment (justinsalamon) [inline docs/installation.rst:15]: "On Linux/macOS SoX is replaced by SoxBindings which is significantly faster, giving better overall performance in Scaper. On these platforms SoxBindings is installed automatically when calling pip install scaper (see below)."

**Review by justinsalamon — CHANGES_REQUESTED** [review]
Couple quick changes

**Review by justinsalamon — APPROVED** [review]
One last thing - can we replace "overall performance" to "runtime performance" ?  I think that's clearer.

**Pull request merged**